backend/tests/accounts/test_views.py

```diff
@@ -413,6 +414,66 @@ def test_update_same_major_twice(self):
         self.assertEqual(response.status_code, 200)
 
 
+class ProfilePicViewTestCase(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            pennid=1,
+            username="student",
+            password="secret",
+            first_name="First",
+            last_name="Last",
+            email="test@test.com",
+        )
+
+        self.client = APIClient()
+        self.serializer = UserSerializer(self.user)
+
+    def test_profile_pic_upload_empty(self):
+        # empty image throws an error
+        resp = self.client.post(reverse("accounts:me-pfp-upload"))
+        self.assertIn(resp.status_code, [400, 403], resp.content)
+
+    def test_profile_pic_upload_success(self):
+        # successful image upload
+        self.client.force_authenticate(user=self.user)
+        resp = self.client.post(
+            reverse("accounts:me-pfp-upload"),
+            {
+                "profile_pic": open(
+                    os.path.join(os.getcwd(), "tests", "accounts", "test_pfp.jpg"),
+                    "rb",
+                )
+            },
+        )
+        self.assertIn(resp.status_code, [200, 201], resp.content)
+
+        # ensure image url is set
+        resp = self.client.get(reverse("accounts:me"))
+        self.assertIn(resp.status_code, [200, 204], resp.content)
+        data = json.loads(resp.content.decode("utf-8"))
+        self.assertTrue(data["profile_pic"])
+
+    def test_profile_pic_upload_non_image(self):
+        # non-image upload should fail
+        self.client.force_authenticate(user=self.user)
```

Comment:
Hmm the fact that it says `force_authenticate` seems a bit sketch, but I don't have full context on what this does. Also, this is a test.

What does this test do?

Reply:
`force_authenticate` is a method provided by Django REST framework for authentication during testing (see here https://www.django-rest-framework.org/api-guide/testing/#forcing-authentication). I saw that other tests we have use it so I used it here—I think it's just so that we don't have to go through the process of actually generating auth credentials for tests since we're not testing auth.